test/statement.test.ts

```diff
@@ -0,0 +1,532 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
```

Comment:
Like in the source, this tests is the workhorse test file of the suite. It is responsible for most of the coverage across the SDK. We have multiple tests for edge cases and specific expected behavior, as well as regression style tests pulled from past SDK experience.